streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,39 +142,31 @@ public StreamTask(final TaskId id,
         log.info("{} Initializing state stores", logPrefix);
         initializeStateStores();
         stateMgr.registerGlobalStateStores(topology.globalStateStores());
-        initTopology();
+        init();
         processorContext.initialized();
     }
 
-    /**
-     * Adds records to queues. If a record has an invalid (i.e., negative) timestamp, the record is skipped
-     * and not added to the queue for processing
-     *
-     * @param partition the partition
-     * @param records  the records
-     * @return the number of added records
-     */
-    @SuppressWarnings("unchecked")
-    public int addRecords(TopicPartition partition, Iterable<ConsumerRecord<byte[], byte[]>> records) {
-        final int oldQueueSize = partitionGroup.numBuffered();
-        final int newQueueSize = partitionGroup.addRawRecords(partition, records);
-
-        log.trace("{} Added records into the buffered queue of partition {}, new queue size is {}", logPrefix, partition, newQueueSize);
-
-        // if after adding these records, its partition queue's buffered size has been
-        // increased beyond the threshold, we can then pause the consumption for this partition
-        if (newQueueSize > this.maxBufferedSize) {
-            consumer.pause(singleton(partition));
+    @Override
+    public void init() {
+        // initialize the task by initializing all its processor nodes in the topology
+        log.info("{} Initializing processor nodes of the topology", logPrefix);
+        for (final ProcessorNode node : topology.processors()) {
+            processorContext.setCurrentNode(node);
+            try {
+                node.init(processorContext);
+            } finally {
+                processorContext.setCurrentNode(null);
+            }
         }
-
-        return newQueueSize - oldQueueSize;
     }
 
     /**
-     * @return The number of records left in the buffer of this task's partition group
+     * re-initialize the task
      */
-    public int numBuffered() {
-        return partitionGroup.numBuffered();
+    @Override
+    public void resume() {
```

Comment:
I thought that 

1. `init` include both initializing the topology (i.e. processorNode.init) as well as the state stores; 
2. and `suspend` does not close state stores, only flush them.
3. and `resume` only re-init the topology but not state stores.

cc @mjsax ?